core/src/main/scala/org/apache/spark/util/SizeEstimator.scala

```diff
@@ -97,6 +98,11 @@ private[spark] object SizeEstimator extends Logging {
       return System.getProperty("spark.test.useCompressedOops").toBoolean
     }
 
+    // java.vm.info provides compressed ref info for IBM JDKs
+    if (System.getProperty("java.vendor").contains("IBM")) {
```

Comment:
If I merge your other change that introduces an enum for JVM vendor, can it be used here?